includes/class-wc-coupon.php

```diff
@@ -721,8 +721,8 @@ public function increase_usage_count( $used_by = '' ) {
 	 */
 	public function decrease_usage_count( $used_by = '' ) {
 		if ( $this->get_id() && $this->get_usage_count() > 0 && $this->data_store ) {
-			$this->set_prop( 'usage_count', ( $this->get_usage_count( 'edit' ) - 1 ) );
-			$this->data_store->decrease_usage_count( $this, $used_by );
+			$new_count = $this->data_store->decrease_usage_count( $this, $used_by );
+			$this->set_prop( 'usage_count', $new_count );
```

Comment:
May be an issue here - if this object is 'saved', won't this increment or decrement be ignored when the data-store runs the update logic? This should probably bypass set_prop and not be part of `changes`